subsys/net/ip/net_shell.c

```diff
@@ -2734,46 +2773,98 @@ static inline void remove_ipv4_ping_handler(void)
 }
 
 static enum net_verdict handle_ipv4_echo_reply(struct net_pkt *pkt,
-						struct net_ipv4_hdr *ip_hdr)
+						struct net_ipv4_hdr *ip_hdr,
+						struct net_icmp_hdr *icmp_hdr)
 {
-	PR_SHELL(shell_for_ping, "Received echo reply from %s to %s\n",
+	NET_PKT_DATA_ACCESS_CONTIGUOUS_DEFINE(icmp_access,
+					      struct net_icmpv4_echo_req);
+	u32_t cycles;
+	struct net_icmpv4_echo_req *icmp_echo;
+
+	icmp_echo = (struct net_icmpv4_echo_req *)net_pkt_get_data(pkt, &icmp_access);
+	if (icmp_echo == NULL) {
+		return -NET_DROP;
+	}
+
+	net_pkt_skip(pkt, sizeof(*icmp_echo));
+	if (net_pkt_read_be32(pkt, &cycles)) {
+		return -NET_DROP;
+	}
+
+	cycles = k_cycle_get_32() - cycles;
+
+	PR_SHELL(shell_for_ping, "%d bytes from %s to %s: icmp_seq=%d ttl=%d time=%.2f ms\n",
```

Comment:
Print float data depends on some additional configs, otherwise, the time can not output correctly.

28 bytes from 192.168.1.1 to 192.168.1.100: icmp_seq=1 ttl=64 time=%.2f ms

Reply:
I suppose it's `CONFIG_NEWLIB_LIBC_FLOAT_PRINTF` and it only affects the formatting to two significant digits as `%f` is used elsewhere in that file.

Does it work for you now?

Reply:
It also doesn't work with %f, can you please use %d instead as this is a universal network tool.

28 bytes from 192.168.1.1 to 192.168.1.100: icmp_seq=0 ttl=64 time=%f ms

Reply:
Ok.

Reply:
What target / config are you using?
When I build for nrf52840_pca10056, `CONFIG_NEWLIB_LIBC_FLOAT_PRINTF` is not set, yet `%.2f` is available.

Reply:
@benpicco I'm using 96boards IVY5661, the board support patch(https://github.com/zephyrproject-rtos/zephyr/pull/10626) is not merged yet.

I tried to store time in a variable before output, It's very strange that printf works, but PR_SHELL not. It seems there is some issue with PR_SHELL.
```
    float time;
    time =  SYS_CLOCK_HW_CYCLES_TO_NS(cycles) / 1000000.f;
    printf("time: %f.\n", time);

    PR_SHELL(shell_for_ping, "%d bytes from %s to %s: icmp_seq=%d ttl=%d time=%f ms\n",
         ntohs(ip_hdr->len) - net_pkt_ipv6_ext_len(pkt) - NET_ICMPH_LEN,
         net_sprint_ipv4_addr(&ip_hdr->src),
         net_sprint_ipv4_addr(&ip_hdr->dst),
         ntohs(icmp_echo->sequence),
         ip_hdr->ttl,
         time);
```
OUTPUT:
time: 4.418798.
28 bytes from 192.168.1.1 to 192.168.1.101: icmp_seq=0 ttl=64 time=%f ms